### DISCUSSION TITLE
Cleanup actions on workflow instance cancellation

### DISCUSSION THREAD

**Discussion by martinmanfar** [elsa-workflows/elsa-core#5520] [Q&A]
Hi,

When I cancel a workflow instance I have to do some cleanup actions in the current suspended activities. However I cannot find any signal or event to register. Do you how could I do it please?

Thanks!

**Comment by glime-ai** [comment]
To cancel a workflow instance and perform cleanup actions on the current suspended activities, you can follow these steps:

1. **Cancel the Workflow Instance**: You can cancel a workflow instance by setting its status to 'Cancelled'. This will prevent the workflow from being resumed.

2. **Perform Cleanup Actions**: To perform cleanup actions, you would typically need to interact with the workflow instance's data, such as bookmarks, blocking activities, and scheduled activities. You can do this by accessing the workflow instance's data from the database or the workflow context and then executing your cleanup logic.

3. **Listen for Cancellation Events**: Elsa does not currently provide a built-in event that fires when a workflow instance is canceled. However, you can extend Elsa's functionality by creating custom activities or services that check the workflow instance's status and perform the necessary cleanup when they detect that the instance has been canceled.

4. **Custom Suspension Strategy**: If you need more control over the suspension and resumption of activities, you might consider implementing a custom suspension strategy. This would involve writing custom code that handles the suspension and resumption of activities based on your specific requirements.

5. **Check for Elsa Updates**: Since Elsa is an actively developed project, it's a good idea to check the official documentation and GitHub repository for any updates or new features that might provide the functionality you're looking for.

Remember to handle exceptions and edge cases in your cleanup logic to ensure that the cleanup is performed reliably.

If you need further assistance with implementing a custom solution, you may want to reach out to the Elsa community or check the official documentation for guidance.

This comment was generated by [Glime](https://glimelab.ai/?utm_source=dRbvTb7IhCSYiI&utm_medium=github).
